### PR TITLE
GEA 4790 updates

### PR DESCRIPTION
**Pull request by pangea-andrest**

### CODE REVIEW

File: packages/pangea-sdk/tests/integration/test_vault.py [hunk 1]
```diff
@@ -523,68 +375,164 @@ def jwt_signing_cycle(self, id):
...
-        # Revoke key
-        revoke_resp = self.vault.revoke(id)
-        self.assertEqual(id, revoke_resp.result.id)
+        # Deactivate key
+        state_change_resp = self.vault.state_change(id, ItemVersionState.DEACTIVATED, version=1)
+        self.assertEqual(id, state_change_resp.result.id)
 
         # Verify after revoked.
         verify1_revoked_resp = self.vault.jwt_verify(jws_v1)
         self.assertTrue(verify1_revoked_resp.result.valid_signature)
```
Comment (dulanowiczpangea) [inline packages/pangea-sdk/tests/integration/test_vault.py:384]: We are no longer using revoke.

File: packages/pangea-sdk/pangea/services/vault/models/common.py [hunk 2]
```diff
@@ -148,40 +170,47 @@ class GetRequest(APIRequestModel):
...
+    state: str
+    created_at: str
+    public_key: Optional[EncodedPublicKey] = None
+    secret: Optional[str] = None
+
+
+class GetResult(PangeaResponseResult):
     type: str
     id: str
-    version: int
```
Comment (dulanowiczpangea) [inline packages/pangea-sdk/pangea/services/vault/models/common.py:154]: Please rename "version" to "current_version" (int).

File: packages/pangea-sdk/pangea/services/vault/models/common.py [hunk 3]
```diff
@@ -148,40 +170,47 @@ class GetRequest(APIRequestModel):
...
-    last_rotated: Optional[str] = None  # TODO: should be time
-    next_rotation: Optional[str] = None  # TODO: should be time
-    expiration: Optional[str] = None  # TODO: should be time
-    rotation_policy: Optional[str] = None
+    last_rotated: Optional[str] = None
+    next_rotation: Optional[str] = None
+    expiration: Optional[str] = None
+    rotation_frequency: Optional[str] = None
     identity: str
     version: int
```
Comment (dulanowiczpangea) [inline packages/pangea-sdk/pangea/services/vault/models/common.py:207]: Please change "version" to "versions" object according to new specs.

File: packages/pangea-sdk/pangea/services/vault/models/common.py [hunk 4]
```diff
@@ -148,40 +170,47 @@ class GetRequest(APIRequestModel):
...
     name: Optional[str] = None
     folder: Optional[str] = None
     metadata: Optional[Metadata] = None
     tags: Optional[Tags] = None
-    created_at: str  # TODO: should be time
-    revoked_at: Optional[str] = None  # TODO: should be time
+    created_at: str
+    destroyed_at: Optional[str] = None
```
Comment (dulanowiczpangea) [inline packages/pangea-sdk/pangea/services/vault/models/common.py:214]: Please add "state" and "has_compromised_versions" fields.

File: packages/pangea-sdk/pangea/services/vault/models/common.py [hunk 5]
```diff
@@ -129,59 +153,59 @@ class CommonGenerateRequest(APIRequestModel):
...
-    version: Optional[int] = None
+    version: Optional[Union[str, int]] = None
     verbose: Optional[bool] = None
+    version_state: Optional[ItemVersionState] = None
 
 
-class CommonGetResult(PangeaResponseResult):
-    type: str
-    id: str
+class ItemVersionData(PangeaResponseResult):
```
Comment (dulanowiczpangea) [inline packages/pangea-sdk/pangea/services/vault/models/common.py:174]: Please remove public_key and secret from this object.
Also you could rename the object as CurrentVersionData.

File: packages/pangea-sdk/pangea/services/vault/models/common.py [hunk 6]
```diff
@@ -129,59 +153,59 @@ class CommonGenerateRequest(APIRequestModel):
...
+    purpose: Optional[str] = None
+
+
+class GetResult(ItemData):
+    versions: List[ItemVersionData] = []
+    rotation_grace_period: Optional[str] = None
+
+
+class ListItemData(ItemData):
+    compromised_versions: List[ItemVersionData]
```
Comment (dulanowiczpangea) [inline packages/pangea-sdk/pangea/services/vault/models/common.py:208]: Where are all the missing fields?

	ID                string            `json:"id,omitempty"`
	CreatedAt         time.Time         `json:"created_at,omitempty"`
	Tags              []string          `json:"tags,omitempty"`
	Metadata          pangeadb.Metadata `json:"metadata,omitempty"`
	LastRotated       *time.Time        `json:"last_rotated,omitempty"`
	NextRotation      *time.Time        `json:"next_rotation,omitempty"`
	Expiration        *time.Time        `json:"expiration,omitempty"`
	RotationFrequency *string           `json:"rotation_frequency,omitempty"`
	RotationState     *string           `json:"rotation_state,omitempty"`
	ItemState         string            `json:"item_state"`

	// only valid for keys
	Algorithm *string `json:"algorithm,omitempty"`
	Purpose   *string `json:"purpose,omitempty"`

	CurrentVersion      *ListResponseItemVersion   `json:"current_version"`
	CompromisedVersions []*ListResponseItemVersion `json:"compromised_versions"`

Comment (pangea-andrest) [reply packages/pangea-sdk/pangea/services/vault/models/common.py:208]: They are on ItemData

File: packages/pangea-sdk/pangea/services/vault/models/common.py (hunk 3 above)
Comment (pangea-andrest) [reply packages/pangea-sdk/pangea/services/vault/models/common.py]: fixed

File: packages/pangea-sdk/tests/integration/test_vault.py (hunk 1 above)
Comment (pangea-andrest) [reply packages/pangea-sdk/tests/integration/test_vault.py]: fixed

File: packages/pangea-sdk/pangea/services/vault/models/common.py (hunk 5 above)
Comment (pangea-andrest) [reply packages/pangea-sdk/pangea/services/vault/models/common.py:174]: Aren't they returned when you do a get? they are optional values

File: packages/pangea-sdk/pangea/services/vault/models/common.py [hunk 7]
```diff
@@ -129,59 +153,59 @@ class CommonGenerateRequest(APIRequestModel):
...
-    retain_previous_version: Optional[bool] = None
+    state: str
+    created_at: str
+    destroy_at: Optional[str] = None
+    public_key: Optional[EncodedPublicKey] = None
+    secret: Optional[str] = None
 
 
-class ListItemData(APIRequestModel):
+class ItemData(PangeaResponseResult):
```
Comment (dulanowiczpangea) [inline packages/pangea-sdk/pangea/services/vault/models/common.py:183]: There are mixed fields in this object from verbose (Item.Type != "folder") with not verbose (Item.Type == "folder").

File: packages/pangea-sdk/pangea/services/vault/models/common.py (hunk 6 above)
Comment (dulanowiczpangea) [reply packages/pangea-sdk/pangea/services/vault/models/common.py:208]: Ok

File: packages/pangea-sdk/pangea/services/vault/models/common.py (hunk 5 above)
Comment (pangea-andrest) [reply packages/pangea-sdk/pangea/services/vault/models/common.py:174]: that's when request /list, but when request /get you also get secret or public_key. I'm using same object with optional fields

File: packages/pangea-sdk/pangea/services/vault/models/common.py (hunk 7 above)
Comment (pangea-andrest) [reply packages/pangea-sdk/pangea/services/vault/models/common.py:183]: Fixed

File: packages/pangea-sdk/pangea/services/vault/models/common.py (hunk 5 above)
Comment (dulanowiczpangea) [reply packages/pangea-sdk/pangea/services/vault/models/common.py:174]: Ok

File: packages/pangea-sdk/pangea/services/vault/models/common.py [hunk 8]
```diff
@@ -125,63 +149,63 @@ class CommonStoreResult(PangeaResponseResult):
...
+    purpose: Optional[str] = None
+
+
+class GetResult(ItemData):
+    versions: List[ItemVersionData] = []
+    rotation_grace_period: Optional[str] = None
+
+
+class ListItemData(ItemData):
+    compromised_versions: List[ItemVersionData]
```
Comment (dulanowiczpangea) [inline packages/pangea-sdk/pangea/services/vault/models/common.py:208]: In /list endpoint when Type == "folder" you must not show compromised_versions field. This field should be Optional. Also add the corresponding logic to the endpoint.

		if respItem.Type != "folder" {
			respItem.ListResponseItemVerbose = &ListResponseItemVerbose{
				ID:                  item.ID,
				ItemState:           item.State,
				CreatedAt:           item.CreatedAt,
				Expiration:          item.Expiration,
				Tags:                item.Tags.Strings(),
				Metadata:            item.Metadata,
				RotationFrequency:   item.RotationFrequency,
				RotationState:       item.RotationState,
				LastRotated:         item.LastRotated,
				NextRotation:        item.NextRotation,
				Algorithm:           item.Algorithm,
				Purpose:             item.Purpose,
				CurrentVersion:      currentVersion[item.ID],
				CompromisedVersions: compromised,
			}
		}

**Pull request merged**